src/DssSpell.sol

```diff
@@ -49,82 +121,187 @@ contract DssSpellAction is DssAction {
-        _updateDoc("RWA007-A", "QmejL1CKKN5vCwp9QD1gebnnAM2MJSt9XbF64uy4ptkJtR");
-        // SG Forge OFH (RWA008-A) legal update doc
-        _updateDoc("RWA008-A", "QmZ4heYjptvj3ovafADJpXYMFXMyY3yQjkTXpvjFPnAKcy");
-        // HVB (RWA009-A) legal update doc
-        _updateDoc("RWA009-A", "QmeRrbDF8MVPQfNe83gWf2qV48jApVigm1WyjEtDXCZ5rT");
-
-        // RWA007-A autoline changes:
-        // - bump oracle price to 500m
-        // - increase DC to 500m
-        // - increase autoline `gap` to 100m
-        //
-        // https://vote.makerdao.com/polling/QmSfMtTM#poll-detail
-        DssExecLib.setIlkAutoLineParameters(
-            "RWA007-A",
-            500 * MILLION,
-            100 * MILLION,
-            1 weeks
-        );
-        RwaLiquidationLike(MIP21_LIQUIDATION_ORACLE).bump(
-            "RWA007-A",
-             500 * MILLION * WAD
-        );
-        DssExecLib.updateCollateralPrice("RWA007-A");
-
-        // RETH-A autoline changes:
-        // - line 5m
-        // - gap 3m
-        // - ttl 8h
-        //
-        //  https://vote.makerdao.com/polling/QmfMswF2#poll-detail
-        DssExecLib.setIlkAutoLineParameters(
-            "RETH-A",
-            5 * MILLION,
-            3 * MILLION,
-            8 hours
-        );
-
-        // -------------------- Changelog Update & housekeeping ---------------------
-
-        // - Change "RWA007_A_INPUT_CONDUIT_URN" to "RWA007_A_INPUT_CONDUIT"
-        // - Change "RWA007_A_INPUT_CONDUIT_JAR" to "RWA007_A_JAR_INPUT_CONDUIT"
-        //
-        DssExecLib.setChangelogAddress("RWA007_A_INPUT_CONDUIT", DssExecLib.getChangelogAddress("RWA007_A_INPUT_CONDUIT_URN"));
-        DssExecLib.setChangelogAddress("RWA007_A_JAR_INPUT_CONDUIT", DssExecLib.getChangelogAddress("RWA007_A_INPUT_CONDUIT_JAR"));
-        ChainlogLike(DssExecLib.LOG).removeAddress("RWA007_A_INPUT_CONDUIT_URN");
-        ChainlogLike(DssExecLib.LOG).removeAddress("RWA007_A_INPUT_CONDUIT_JAR");
-
-        // Bump version
-        DssExecLib.setChangelogVersion("1.14.5");
+            D3MMomLike(D3M_MOM).setAuthority(DssExecLib.getChangelogAddress("MCD_ADM"));
+
+            D3MCompoundPoolLike(D3M_COMPOUND_POOL).file("king", address(this));
+
+            D3MCompoundPlanLike(D3M_COMPOUND_PLAN).rely(D3M_MOM);
+            D3MCompoundPlanLike(D3M_COMPOUND_PLAN).file("barb", D3M_COMP_BORROW_RATE);
+
+            D3MOracleLike(D3M_ORACLE).file("hub", D3M_HUB);
+
+            spot.file(ILK, "pip", D3M_ORACLE);
+            spot.file(ILK, "mat", RAY);
+            vat.rely(D3M_HUB);
+            vat.init(ILK);
+            JugAbstract(DssExecLib.jug()).init(ILK);
+            DssExecLib.increaseGlobalDebtCeiling(5 * MILLION);
+            DssExecLib.setIlkDebtCeiling(ILK, 5 * MILLION);
+            DssExecLib.setIlkAutoLineParameters(ILK, 5 * MILLION, 5 * MILLION, 12 hours);
+            DssExecLib.updateCollateralPrice(ILK);
+
+            // Add to ilk registry
+            IlkRegistryAbstract(DssExecLib.reg()).put(
```

Comment:
Deferring this to @brianmcmichael not sure about the class type.

Reply:
I've also made a couple of questions in the test section regarding the`gem` and `pip`. We might want to fill them.

Reply:
The prev D3M is class type 1, but it has a clipper so not sure what to do here.

Reply:
Clippers don't really make sense for D3M so I'm treating it like teleport. Previously we added clippers just for convenience. I don't think we need to here.

Reply:
Yeah I kind of agree we shouldn't be adding too much overhead but I was more meaning that class type == 1 meant the ones that had a specific auctions contract.

Reply:
Yeah probably not a type 4. Did we put the aave d3m into the registry?

Reply:
`"DIRECT-AAVEV2-DAI"` >> `0x4449524543542d4141564556322d444149`

is a Class 1, so probably not a good example.

I'd reserve Class 4 for traditional collateral types. Maybe pick a different number space here, like `101`, we'll probably need to document whatever we pick either in the README for ilk-registry.

Reply:
We used 4 for Teleport, so not sure what is the best here...